### PR TITLE
Issue/8914 async posting continue editing

### LINKED ISSUE #8914: Media Async: Restore attachment state when coming back to a post that is marked for async upload

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #8914 

This PR allows the Editor interface to restore the attachment upload interface when going back to post that are marked for async upload.

For non async posting it also improves the code for connecting back attachment when switching between HTML mode and Visual while uploads are ongoing.

To test:
 - Start a new post
 - Add new media to it
 - While the media is uploading tap to async upload
 - While the upload of media are still going, open back the editor and see if they are being processed correctly
 - Try the same by just switching between HTML and Visual mode while uploads are ongoing.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 1]
```diff
@@ -2694,38 +2736,9 @@ extension AztecPostViewController {
         refreshNavigationBar()
     }
 
-
     fileprivate func observe(media: Media) {
         let _ = mediaCoordinator.addObserver({ [weak self](media, state) in
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: What happens when you add new Media to the post? by using insert? will the post observer update with the new media inserted?

**Comment by SergioEstevao** [comment]
> So I tested this a couple of times, and sometimes the reconnection worked and sometimes it didn't. The last two times I came back to the editor, the navigation bar progress continued, but the media never reconnected:

Humm, it looks for some reason the original image was not inserted correctly before the upload method was called. The image needs to have a reference to the core data id of media. It looks it got empty out for some reason.

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (frosty) [reply WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: The post observer should update, as long as the relationship between media and post is present. I just tested by adding logging into the `mediaObserver(media: Media, state: MediaCoordinator.MediaState)` method, and it logged everything twice.

**Comment by SergioEstevao** [comment]
Ready for another look @frosty

**Comment by frosty** [comment]
Hmm, I just tried it a second time and the progress UI appeared a lot quicker. Not quite sure what's happening.

File: WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift [hunk 2]
```diff
@@ -2855,7 +2849,7 @@ extension AztecPostViewController {
             attachment.uploadID = media.uploadID
             let info = MediaAnalyticsInfo(origin: .editor(.wpMediaLibrary), selectionMethod: mediaSelectionMethod)
             mediaCoordinator.addMedia(media, analyticsInfo: info)
-            observe(media: media)
+            media.addPostsObject(post)
```
[the remark this answers is not in the document]
Comment (SergioEstevao) [reply WordPress/Classes/ViewRelated/Aztec/ViewControllers/AztecPostViewController.swift]: Good idea!

**Comment by SergioEstevao** [comment]
@frosty ready for another look.

**Pull request merged**